### PR TITLE
Write quick start guide for Timer.

### PR DESCRIPTION
**Pull request by robieta**
The focus is mainly on the instruction counting APIs, though there are some wall time bits as well.

### CODE REVIEW

**Comment by robieta** [comment]
CC @ngimel @ezyang @brianjo @jspisak 

I'm working on a parallel one in the main repo for `Measurement`, `CallgrindStats`, and `FunctionCounts` docstrings. (That one should be simpler.)

**Comment by robieta** [comment]
Looks like I messed up the formatting. (Naturally.) Fixing now.

File: recipes_source/recipes/timer_quick_start.py [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+#
+#     646262
+#     646262
+#     646262
+#
+
+###############################################################################
+# As you can see, one of the selling points of Callgrind is that it is 100%
+# deterministic. This is very useful because it lets us A/B test very subtle
+# changes. Let's take a look at the CallgrindStats object that Timer returns:
```
Comment (ngimel) [inline recipes_source/recipes/timer_quick_start.py:134]: Does it make sense to mention that you need REL_WITH_DEB_INFO for detailed counts info? A warning will tell you if you don't have it, but still, having it here also makes sense to me.

File: recipes_source/recipes/timer_quick_start.py [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+#         100  /data/users/taylorrobie/repos/pytorch/build/../aten/src/ATen/core/dispatch/DispatchKeyExtract ... r (at::Tensor const&, at::Tensor const&)> const&, at::Tensor const&, at::Tensor const&) const
+#         100  /data/users/taylorrobie/repos/pytorch/build/../aten/src/ATen/core/boxing/impl/WrapFunctionInt ... const&, at::Tensor const&)>::call(c10::OperatorKernel*, at::Tensor const&, at::Tensor const&)
+#
+#     Total: 587972
+#
+
+###############################################################################
+# There's some useful information here. The most expensive sections of the code
+# are malloc, free, and various PyTorch utilities. However there is also a lot
+# of extraneous path junk cluttering things up. For that, you can call the
```
Comment (ngimel) [inline recipes_source/recipes/timer_quick_start.py:215]: junk? Also, probably explain what `as_standardized` does right away, and don't show output without it.

**Comment by robieta** [comment]
Looks like linewidth=200 is far too large for the website so the Callgrind part is ugly.

I'm also going to adjust the structure based on some high level comments from @ezyang.

File: recipes_source/recipes/timer_quick_start.py [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+#         100  /usr/include/c++/8/bits/atomic_base.h:void at::native::DispatchStub<void (*)(at::TensorIterator&), at::native::mul_stub>::operator()<at::TensorIterator&>(c10::DeviceType, at::TensorIterator&)
+#         100  /usr/include/c++/8/bits/atomic_base.h:c10::TensorImpl::~TensorImpl()
+#         100  /usr/include/c++/8/bits/atomic_base.h:at::native::mul(at::Tensor const&, at::Tensor const&)
+#         100  /home/taylorrobie/local/miniconda3/envs/ab_ref/lib/python3.7/site-packages/torch/include/c10/util/intrusive_ptr.h:c10::intrusive_ptr_target::release_resources()
+#
+#     Total: 587972
+#
+
+###############################################################################
+# `CallgrindStats.stats()` produces a `FunctionCounts` object, which is really
```
Comment (ngimel) [inline recipes_source/recipes/timer_quick_start.py:250]: this explanation belongs above, where you are first calling `stats()`?

File: recipes_source/recipes/timer_quick_start.py [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+#
+
+###############################################################################
+# As you can see, one of the selling points of Callgrind is that it is 100%
+# deterministic. This is very useful because it lets us A/B test very subtle
+# changes. Let's take a look at the CallgrindStats object that Timer returns:
+
+from torch.utils.benchmark import CallgrindStats
+
+s0: CallgrindStats = Timer(
```
Comment (ngimel) [inline recipes_source/recipes/timer_quick_start.py:138]: for people uninitiated to the benefits of type checking, this syntax may be surprising, and deserves a comment

File: recipes_source/recipes/timer_quick_start.py [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+#         100  /usr/include/c++/8/bits/atomic_base.h:c10::TensorImpl::~TensorImpl()
+#         100  /usr/include/c++/8/bits/atomic_base.h:at::native::mul(at::Tensor const&, at::Tensor const&)
+#         100  /home/taylorrobie/local/miniconda3/envs/ab_ref/lib/python3.7/site-packages/torch/include/c10/util/intrusive_ptr.h:c10::intrusive_ptr_target::release_resources()
+#
+#     Total: 587972
+#
+
+###############################################################################
+# `CallgrindStats.stats()` produces a `FunctionCounts` object, which is really
+# just a glorified tuple. Both CallgrindStats and FunctionCounts are
```
Comment (ngimel) [inline recipes_source/recipes/timer_quick_start.py:251]: glorified tuple consisting of what?

File: recipes_source/recipes/timer_quick_start.py (hunk 1 above)
Comment (robieta) [reply recipes_source/recipes/timer_quick_start.py]: I've added a footnote section and included it.

File: recipes_source/recipes/timer_quick_start.py (hunk 2 above)
Comment (robieta) [reply recipes_source/recipes/timer_quick_start.py]: Done. (And yes, not a great way of phrasing.)

File: recipes_source/recipes/timer_quick_start.py (hunk 3 above)
Comment (robieta) [reply recipes_source/recipes/timer_quick_start.py]: Done.

File: recipes_source/recipes/timer_quick_start.py (hunk 4 above)
Comment (robieta) [reply recipes_source/recipes/timer_quick_start.py]: Done.

File: recipes_source/recipes/timer_quick_start.py (hunk 5 above)
Comment (robieta) [reply recipes_source/recipes/timer_quick_start.py]: I've reworded this.

**Comment by robieta** [comment]
I've adjusted the text and done some layout improvements. PTAL.

**Review by ngimel — APPROVED** [review]
This looks great, sorry for delay!

**Comment by robieta** [comment]
@brianjo Would you be so kind as to merge this? (I don't have permission.)

**Pull request merged**